test/integration/jobset_controller_test.go

```diff
@@ -188,6 +189,30 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 			// Check JobSet has completed.
 			gomega.Eventually(checkJobSetStatus, timeout, interval).WithArguments(js, jobset.JobSetCompleted).Should(gomega.Equal(true))
 		})
+
+		ginkgo.It("should throw an error if job completion mode is not indexed", func() {
```

Comment:
Actually once we have validation, this test will need to change. 

I think we should add this test once we add the validation on the field; ideally using CEL:

```
// +kubebuilder:validation:XValidation:rule
```

Reply:
Added CEL validation rule and updated integration test case.